ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -48,114 +30,151 @@ let update_mh_info interface =
   in
   ()
 
-let _restart_stunnel_no_cache ~__context ~accept =
-  let (_ : Thread.t) =
-    Thread.create (fun () -> Helpers.Stunnel.restart ~__context ~accept) ()
-  in
-  ()
+module Stunnel = struct
+  let stunnel_accept_m = Mutex.create ()
+
+  let stunnel_accept = ref None
+
+  let _restart_stunnel_no_cache ~__context ~accept =
+    let (_ : Thread.t) =
+      Thread.create (fun () -> Helpers.Stunnel.restart ~__context ~accept) ()
+    in
+    ()
+
+  let restart_stunnel ~__context ~accept =
+    info "Restarting stunnel (accepting connections on %s)" accept ;
+    (* cache `accept` so client can call `reconfigure_stunnel` easily *)
+    Mutex.execute stunnel_accept_m (fun () -> stunnel_accept := Some accept) ;
+    _restart_stunnel_no_cache ~__context ~accept
+
+  let reconfigure_stunnel ~__context =
+    let f =
+      Mutex.execute stunnel_accept_m (fun () ->
+          match !stunnel_accept with
+          | None ->
+              fun () ->
+                D.warn
+                  "reconfigure_stunnel: accept is not set, so not restarting \
+                   stunnel"
+          | Some accept ->
+              fun () -> _restart_stunnel_no_cache ~__context ~accept)
+    in
+    f ()
+end
+
+module Server = struct
+  (* Keep track of the management interface server thread.
+     Stores a key into the table in Http_srv which identifies the server thread bound
+     to the management IP. *)
+  let management_servers = ref []
+
+  let stop () =
+    debug "Shutting down the old management interface (if any)" ;
+    List.iter (fun i -> Http_svr.stop i) !management_servers ;
+    management_servers := []
+
+  (* Even if xapi listens on all IP addresses, there is still an interface appointed as
+     _the_ management interface. Hosts in a pool use the IP addresses of this interfaces
+     to communicate with each other. *)
+  let start ~__context ?addr () =
+    let socket, accept =
+      match addr with
+      | None -> (
+          info "Starting new server (listening on all IP addresses)" ;
+          try
+            (* Is it IPv6 ? *)
+            let addr = Unix.inet6_addr_any in
+            ( Xapi_http.bind (Unix.ADDR_INET (addr, Constants.http_port))
```

Comment:
It seems strange to me to take the port number from `Constants` but to used a fixed string here at the same time - also below.

Reply:
Indeed, we should use `Constant.https_port` for 443.